### ISSUE TITLE
If a Story table finds a variable match but not a value match, we should warn

### ISSUE THREAD

**Issue by BryceStevenWilley** [plocket/docassemble-cucumber#724]
A variable match but not a value match indicates that there's very likely a misspelling of the value, and that it's a dev error, not that we just aren't able to match. Continuing on might work, but might also cause the interview to be different from what the tester expects, and hides the real issue (which is that they misspelled the value). 

Bonus for dropdowns and choices; list the values that we did see on the screen, and also list which fields were never set by the table.

Some code on an older branch that will need to be re-contextualized before it's useful;

```diff
diff --git a/lib/scope.js b/lib/scope.js
index ae15f03..dc74518 100644
--- a/lib/scope.js
+++ b/lib/scope.js
@@ -1660,10 +1660,11 @@ module.exports = {
   },  // ends setCustomDatatype {}
 
   enter_answer: {
-    button: async function ( scope, { handle, answer }) { await scope.tapElement( scope, handle ); },
+    button: async function ( scope, { handle, answer }) { await scope.tapElement( scope, handle ); return true },
     textarea: async function ( scope, { handle, answer }) {
       await scope.setText( scope, { handle, answer });
       await scope.afterStep(scope);  // No showifs for this one?
+      return true;
     },
     select: async function ( scope, { handle, answer }) {
       // Note: I don't remember what's really going on in here.
@@ -1681,21 +1682,30 @@ module.exports = {
       // values.
       // let option = await handle.$(`option[value="${ answer }"]`);
       let option = await scope.page.$(`option[value="${ answer }"]`);
+      let success = false;
       if ( option ) {
         await handle.select( answer );  // And use that value to set it
+        success = true;
 
       // If that literal value isn't on the page, it should be a base64 encoded value
       // TODO: Can these be double encoded? Can they be encoded like objects?
       } else {
         let base64_name = toBase64( answer );
         await handle.select( base64_name );
+        let base64_option = await scope.page.$(`option[value="${ base64_name }"]`);
+        if (base64_option) {
+          await handle.select( base64_name );
+          success = true;
+        }
       }
       
       await scope.afterStep(scope, { waitForShowIf: true });
+      return success;
     },
     canvas: async function ( scope, { handle, answer }) {
       await scope.draw_signature( scope );
       await scope.afterStep( scope );
+      return true;
     },
     input: async function ( scope, { handle, answer }) {
       /* Set value of some `input` element to the given value. */
@@ -2051,6 +2061,9 @@ module.exports = {
       let num_unused_rows = 0;
       for ( let row of var_data ) {
         if ( row.times_used === 0 ) {
+          if (row.times_var_matched > 0) {
+
+          }
           num_unused_rows += 1;
           let source_row = row;
           // TODO: Make the row object flatter so we don't need to dig in this way
@@ -2102,6 +2115,25 @@ module.exports = {
     // TODO: Since we now continue only once no matter what, we can put
     // the error in the caller of the function, though we may want
     // to pass the symbol to print as well.
+    if ( error.was_found ) {
+      let field_map = {};
+      let trigger_map = {};
+      for (let field of fields) {
+        if (!field.was_set && field.guesses.length > 0) {
+          for (let guess of field.guesses) {
+            if (guess["var"] in field_map) {
+              field_map[guess["var"]].push(guess["value"]);
+            } else {
+              field_map[guess["var"]] = [guess["value"]];
+            }
+            trigger_map[guess["var"]] = field.trigger;
+          }
+        }
+      }
+      for (let field in field_map) {
+        console.log(`didn't set field: ${ JSON.stringify(field) }: ${ JSON.stringify(field_map[field]) } (${ trigger_map[field] })`); 
+      }
+    }
     return error;
     // TODO: Maybe we should return something else or something in addition about used vars
   },  // Ends scope.setFields()

```

**Comment by plocket** [comment]
I think this would be considered a breaking change, as it changes expected behavior and error throwing. As such, it should probably be part of the prep for v5.

**Comment by plocket** [comment]
From our discussions: We should warn instead of error, though we're not sure where we should place the warning.

Edit: One thing we can do is add "WARNING:" to the start of "unused rows". Also, we did start some of this work when we updated the behavior of uploading files by starting to return `row_was_used` when setting variables. We can work more on returning `false` when the values aren't found on the page. See https://github.com/SuffolkLITLab/ALKiln/pull/717/files.
